paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentService.java

```diff
@@ -245,47 +241,47 @@ private RefundPayment createRefundPayment(
-                .eventType(EventType.PAYMENT_PAID)
-                .eventTimestamp(now)
-                .namespace(payment.getNamespace())
-                .paymentId(payment.getRefundPaymentId())
-                .orderId(payment.getOrderId())
-                .userId(payment.getUserId())
-                .paymentPaidTimestamp(now)
-                .orderType(payment.getRefundType());
+        RefundMessage.RefundMessageBuilder refundMessageBuilder = RefundMessage.builder()
+                .eventType(EventType.REFUND_PAID)
+                .timestamp(now)
+                .namespace(refundPayment.getNamespace())
+                .refundId((refundPayment.getRefundId()))
+                .refundPaymentId(refundPayment.getRefundPaymentId())
+                .orderId(refundPayment.getOrderId())
+                .userId(refundPayment.getUserId());
 
-        PaymentMessage paymentMessage = paymentMessageBuilder.build();
+        RefundMessage refundMessage = refundMessageBuilder.build();
 
-        orderPaidWebHookAction(paymentMessage);
-
-        sendEventService.sendEventMessage(TopicName.PAYMENTS, paymentMessage);
-        log.debug("triggered event REFUND_PAID for refundId: " + payment.getRefundPaymentId());
+        refundWebHookAction(refundMessage);
+        log.debug("triggered event REFUND_PAID for refundId: " + refundPayment.getRefundPaymentId());
     }
 
-    protected void orderPaidWebHookAction(PaymentMessage message) {
+    protected void refundWebHookAction(RefundMessage message) {
         try {
-            sendNotificationService.sendPaymentMessageNotification(message);
+            sendNotificationService.sendRefundMessageNotification(message);
         } catch (Exception e) {
             log.error("webhookAction: failed action after receiving event, eventType: " + message.getEventType(), e);
         }
     }
 
-    public void triggerRefundPaymentFailedEvent(Payment payment) {
-        PaymentMessage paymentMessage = PaymentMessage.builder()
-                .eventType(EventType.PAYMENT_FAILED)
-                .namespace(payment.getNamespace())
-                .paymentId(payment.getPaymentId())
-                .orderId(payment.getOrderId())
-                .userId(payment.getUserId())
-                //.paymentPaidTimestamp(payment.getTimestamp())
-                .orderType(payment.getPaymentType())
-                .build();
-        sendEventService.sendEventMessage(TopicName.PAYMENTS, paymentMessage);
-        log.debug("triggered event PAYMENT_FAILED for paymentId: " + payment.getPaymentId());
+    public void triggerRefundPaymentFailedEvent(RefundPayment refundPayment) {
+        String now = DateTimeUtil.getDateTime();
+
+        RefundMessage.RefundMessageBuilder refundMessageBuilder = RefundMessage.builder()
+                .eventType(EventType.REFUND_FAILED)
+                .timestamp(now)
+                .namespace(refundPayment.getNamespace())
+                .refundId(refundPayment.getRefundId())
```

Comment:
Do we have refundPaymentId in here?

Reply:
We should, I'll append it - good that you noticed.